src/Features/Core/Portable/CodeFixes/CodeFixService.cs

```diff
@@ -956,13 +970,45 @@ static ImmutableArray<IConfigurationFixProvider> GetConfigurationFixProviders(Li
                         if (string.IsNullOrWhiteSpace(id))
                             continue;
 
-                        var list = builder.GetOrAdd(id, static _ => new List<CodeFixProvider>());
+                        var list = builder.GetOrAdd(id, static _ => ArrayBuilder<CodeFixProvider>.GetInstance());
                         list.Add(fixer);
                     }
                 }
             }
 
-            return builder.ToImmutable();
+            return builder.ToImmutableDictionary(kvp => kvp.Key, kvp => kvp.Value.ToImmutableAndFree());
+        }
+
+        private sealed class FixerComparer : IComparer<CodeFixProvider>
+        {
+            private readonly Dictionary<CodeFixProvider, int> _fixerToIndex;
+            private readonly ImmutableDictionary<CodeFixProvider, int> _priorityMap;
+
+            public FixerComparer(
+                ArrayBuilder<CodeFixProvider> allFixers,
+                ImmutableDictionary<CodeFixProvider, int> priorityMap)
+            {
+                _fixerToIndex = allFixers.Select((fixer, index) => (fixer, index)).ToDictionary(t => t.fixer, t => t.index);
+                _priorityMap = priorityMap;
+            }
+
+            public int Compare([AllowNull] CodeFixProvider x, [AllowNull] CodeFixProvider y)
+            {
+                Contract.ThrowIfNull(x);
+                Contract.ThrowIfNull(y);
+
+                // If the fixers specify an explicit ordering between each other, then respect that.
+                if (_priorityMap.TryGetValue(x, out var xOrder) &&
+                    _priorityMap.TryGetValue(y, out var yOrder))
```

Comment:
this is teh same issue as https://github.com/dotnet/roslyn/pull/59586/files#diff-5f6cf0794827f40d6b8b8abae0080ef94419f1cffc0d7343a0c21fcf6c6ceb9fR483

Specifically, our tests end up creating a CodeFixService with codefixproviders that have no CodeFixMetadata on them.  So we end up not knowing anything about them and these later bits of code have to be resilient to that.  I don't know if htis is intentional and the real codefixservice is supposed to be able to handle thsi.  Or if this is jsut a case of badly behaving tests doing the wrong thing.